engine/function/src/templates/Numeric.ftl

```diff
@@ -1361,49 +1385,52 @@ public class Numeric {
 
         int n = values.intSize("percentile");
 
-        if (n == 0) {
-            return ${pt.null};
-        }
-
         ${pt.primitive}[] sorted = values.copyToArray();
         Arrays.sort(sorted);
 
         <#if pt.valueType.isFloat >
         if (isNaN(sorted[sorted.length - 1])) {
             return ${pt.boxed}.NaN; // Any NaN will pollute the result and NaN always sorted to the end.
         }
-        </#if>
-
         int nullStart = -1;
         int nullCount = 0;
-
         for (int i = 0; i < n; i++) {
             final ${pt.primitive} val = sorted[i];
             if (val > ${pt.null}) {
                 break; // no more NULL possible
-            } else if (nullStart == -1) {
-                if (isNull(val)) {
+            }
+            if (isNull(val)) {
+                nullCount++;
+                if (nullStart == -1) {
                     nullStart = i;
-                    nullCount++;
                 }
-            } else if (isNull(val)) {
-                nullCount++;
             }
         }
+        <#else>
+        int nullCount = 0;
+        for (int i = 0; i < n && isNull(sorted[i]); i++) {
+            nullCount++;
+            <#if pt.valueType.isFloat >
+            if (nullStart == -1) {
+                nullStart = i;
+            }
+            </#if>
```

Comment:
I'm not convinced this is reachable.

<img width="799" alt="image" src="https://github.com/deephaven/deephaven-core/assets/5250374/415bbef9-0e08-42a9-a797-23a4dfaad643">

Reply:
Good catch, corrected.